src/util.cc

```diff
@@ -100,11 +102,26 @@ static void MakeUtf8String(Isolate* isolate,
                            MaybeStackBuffer<T>* target) {
   Local<String> string;
   if (!value->ToString(isolate->GetCurrentContext()).ToLocal(&string)) return;
+  String::ValueView value_view(isolate, string);
+
+  if (value_view.is_one_byte() && simdutf::validate_ascii(reinterpret_cast<const char*>(value_view.data8()),value_view.length()) ) {
+    target->AllocateSufficientStorage(value_view.length() + 1);
+    target->SetLengthAndZeroTerminate(value_view.length());
+    memcpy(target->out(),
+           reinterpret_cast<const char*>(value_view.data8()),
+           value_view.length());
+    return;
+  }
 
-  size_t storage;
-  if (!StringBytes::StorageSize(isolate, string, UTF8).To(&storage)) return;
-  storage += 1;
+  // Add +1 for null termination.
+  auto storage = simdutf::utf8_length_from_utf16(
+                     reinterpret_cast<const char16_t*>(value_view.data16()),
```

Comment:
Accessing `.data16()` is only valid if `.is_one_byte() == false`

Reply:
@anonrig Possibly, instead of checking that it is ascii and then doing a memcpy, we could do a latin1_to_utf8 conversion?

Reply:
yeah, I just changed the implementation. latin1_to_utf8 seems like the best solution.